src/unistd.rs

```diff
@@ -2785,3 +2785,25 @@ pub fn ttyname(fd: RawFd) -> Result<PathBuf> {
     buf.truncate(nul);
     Ok(OsString::from_vec(buf).into())
 }
+
+/// Get the effective user ID and group ID associated with a Unix domain socket.
+///
+/// See also [getpeereid(3)](https://linux.die.net/man/3/getpeereid)
+#[cfg(any(
+    target_os = "macos",
+    target_os = "ios",
+    target_os = "freebsd",
+    target_os = "openbsd",
+    target_os = "netbsd",
+))]
+pub fn getpeereid(fd: RawFd) -> Result<(Uid, Gid)> {
+    let mut uid = 1;
+    let mut gid = 1;
+
+    let ret = unsafe { libc::getpeereid(fd, &mut uid, &mut gid) };
+    if ret != 0 {
```

Comment:
This could be expressed more simply as
```rust
Errno::result(ret).map((uid, gid))
```

Reply:
I had to write this as 

```rust
Errno::result(ret).map(|_| (Uid(uid), Gid(gid)))
```

to make Rust happy, but that's still nicer than what I had 🙂